File: backend/src/main/java/com/cakeplatform/api/modules/shop/Shop.java
package com.cakeplatform.api.modules.shop;

import com.cakeplatform.api.modules.user.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "shops")
@Data
public class Shop {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "business_name", nullable = false)
    private String businessName;

    private String description;
    private String phone;
    private String email;
    private String address;
    private String city;
    private String state;
    private String pincode;

    @Column(name = "business_category")
    private String businessCategory;

    @Enumerated(EnumType.STRING)
    @Column(name = "business_type")
    private BusinessType businessType;

    @Column(name = "years_in_business")
    private Integer yearsInBusiness;

    @Column(name = "fssai_registration")
    private String fssaiRegistration;

    @Column(name = "address_line_1")
    private String addressLine1;

    @Column(name = "address_line_2")
    private String addressLine2;

    private String area;
    private String district;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "cover_image_url")
    private String coverImageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ShopStatus status = ShopStatus.PENDING;

    @Enumerated(EnumType.STRING)
    @Column(name = "verification_status")
    private VerificationStatus verificationStatus = VerificationStatus.PROCESSING;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true)
    private java.util.List<ShopDeliverySlot> deliverySlots = new java.util.ArrayList<>();
}

File: frontend_v2/types/shop.ts
export type ShopStatus = 'ACTIVE' | 'PENDING' | 'SUSPENDED' | 'REJECTED';

export interface Shop {
  id: number;
  businessName: string;
  businessType?: string;
  businessCategory?: string;
  businessDescription?: string;
  businessPhone?: string;
  businessEmail?: string;
  phone?: string;
  email?: string;
  address?: string;
  addressLine1: string;
  addressLine2?: string;
  area?: string;
  city: string;
  district?: string;
  state: string;
  pincode: string;
  status: ShopStatus;
  verificationStatus?: 'VERIFIED' | 'PROCESSING' | 'UNVERIFIED' | string;
  fssaiRegistration?: string;
  yearsInBusiness?: number;
  rating?: number;
  reviewCount?: number;
  imageUrl?: string;
  bannerUrl?: string;
  coverImageUrl?: string;
  logoUrl?: string;
  isPureVeg?: boolean;
  deliveryAvailable?: boolean;
  deliveryTimeMinutes?: number;
  minOrderValue?: number;
  featuredCategory?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface ShopSearchFilters {
  state?: string;
  district?: string;
  city?: string;
  area?: string;
  businessType?: string;
  search?: string;
  location?: string;
}
